lib/ex_aws/auth.ex

```diff
@@ -97,26 +69,58 @@ defmodule ExAws.Auth do
     |> Keyword.keys
     |> Enum.join(";")
 
+    payload = case body do
+                nil -> "UNSIGNED-PAYLOAD"
+                _ -> ExAws.Auth.Utils.hash_sha256(body)
+              end
```

Comment:
As a minor coding style thing, I would prefer
```
payload = case body do
    nil -> "UNSIGNED-PAYLOAD"
    _ -> ExAws.Auth.Utils.hash_sha256(body)
end
```

This is the style used by the rest of the code base and I would prefer to keep that consistent.